extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-config.js

```diff
@@ -36,7 +63,49 @@ describes.realWin(
 
     it('should build shopping config component', () => {
       expect(() => shoppingConfig.layoutCallback()).to.not.throw();
-      expect(shoppingConfig.isLayoutSupported(Layout.NODISPLAY)).to.be.true;
+    });
+
+    it('throws on no config', async () => {
+      expectAsyncConsoleError(async () => {
+        expect(async () => {
+          await shoppingConfig.buildCallback();
+        }).to.throw(
+          /The amp-story-auto-ads:config should be inside a <script> tag with type=\"application\/json\"​​​/
+        );
+      });
+    });
+
+    it('does use remote config when src attribute is provided', async () => {
+      const exampleURL = 'foo.example';
+      element.setAttribute('src', exampleURL);
+
+      const expectedRemoteResult =
+        '{"city-pop":{"product-tag-id":"city-pop","brand-label":"...","brand-favicon":"...","product-title":"Plastic Love","product-price":"...","product-images":"...","product-details":"...","reviews-page":"...","reviews-data":"...","cta-text":1,"shipping-text":1},"k-pop":{"product-tag-id":"k-pop","brand-label":"...","brand-favicon":"...","product-title":"Gangnam Style","product-price":"...","product-images":"...","product-details":"...","reviews-page":"...","reviews-data":"...","cta-text":1,"shipping-text":1},"eurodance":{"product-tag-id":"eurodance","brand-label":"...","brand-favicon":"...","product-title":"Crystal King Battle","product-price":"...","product-images":"...","product-details":"...","reviews-page":"...","reviews-data":"...","cta-text":1,"shipping-text":1}}';
+
+      expect(
+        JSON.stringify(storeService.get(StateProperty.SHOPPING_DATA))
+      ).to.eql(expectedRemoteResult);
```

Comment:
If you want to compare objects instead of strings you can use `.to.deep.equal`. It should give better error messages.